coordinator/src/metrics.rs

```diff
@@ -243,3 +257,48 @@ fn node_metrics(cx: &Context, inner_node: Arc<ln_dlc_node::node::Node<NodeStorag
         }
     }
 }
+
+pub fn collect_metrics_based_on_node_events(
+    node: Node,
+    mut node_event_receiver: watch::Receiver<Option<Event>>,
+) {
+    tokio::spawn(async move {
+        loop {
+            match node_event_receiver.changed().await {
+                Ok(()) => {
+                    if let Some(Event::PaymentClaimed {
+                        amount_msat,
+                        payment_hash,
+                        ..
+                    }) = node_event_receiver.borrow().clone()
+                    {
+                        let mut conn = match node.pool.get() {
+                            Ok(conn) => conn,
+                            Err(e) => {
+                                tracing::error!("Failed to connect to database during node event post processing event: {e:#}");
+                                continue;
+                            }
+                        };
+
+                        match db::trades::get_by_fee_payment_hash(&mut conn, payment_hash) {
+                            Ok(Some(_)) => {
+                                increment_fee_counter(amount_msat / 1000);
+                            }
+                            Ok(None) => {
+                                // irrelevant, not a fee
+                                continue;
+                            }
+                            Err(e) => {
+                                tracing::error!("Failed to load trades from database in node event post processing: {e:#}");
+                                continue;
+                            }
+                        }
+                    }
+                }
```

Comment:
🤔 The `Ok` arm is all blocking code. Particularly `pool.get` waits until we can get a connection from the pool, which _could_ take a while. I would `spawn_blocking` the whole arm.

You won't be able to keep the `continue`s, but you don't need them anyway because the loop ends after the `match` statement.